### PR TITLE
chore(build): upgrade Go to 1.21.10, use custom builder instead of `cross-builder`

### PR DESCRIPTION
**Pull request by bednar**
Closes https://github.com/influxdata/edge/issues/685

This PR introduces several changes:

1. Upgrades GoLang to `1.21.10`.
2. Upgrades the `protobuf` library to `3.18.3` and standardizes its use across the source base, including in `Dockerfile_build`, `build.sh`, and `test.sh`.

For more info see:

- https://influxdata.slack.com/archives/C5BSZ026L/p1716199474358309
- https://github.com/influxdata/edge/issues/687
- https://protobuf.dev/news/2024-01-31/

### Custom builder

The reason for a new custom builder is compatibility issues between the `protobuf` library and Chronograf's UDFs for Python. The updated version of the `cross-builder` (see [PR #669](https://github.com/influxdata/edge/pull/669)) upgrades `protobuf` to version `26.1`. However, this version introduces breaking changes in the Python protobuf library. The Python build for protobuf, available at [protobuf 5.26.1 on PyPI](https://pypi.org/project/protobuf/5.26.1/), does not support Python 2. As a result, updating to the latest version of protobuf (by using newest `cross-builder) would eliminate support for Python v2 in UDFs.

### CODE REVIEW

**Review by davidby-influx — COMMENTED** [review]
I'm going to defer to @bnpfeife 's review of this...

**Review by bnpfeife — APPROVED** [review]
This looks good to me! It looks like there's no automation around building+pushing the custom builder docker image; however, since it's fairly straightforward to do the build+push from a developer machine, I'm not going to block on that.

Thank you for implementing this! My apologies for being absent from this effort, I've been swamped with several other projects.

File: Dockerfile_build [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    rm /go${GO_VERSION}.linux-${GO_ARCH}.tar.gz
+ENV PATH /usr/local/go/bin:$PATH
+ENV PATH $GOPATH/bin:$PATH
+
+#
+# Install Rust
+#
+# src: https://github.com/influxdata/edge/blob/4677c285014ac27727e5a1ae9bf2c1633afc6ea6/dockerfiles/cross-builder/install-rust.sh#L7
+#
+ENV RUST_LATEST_VERSION=1.63.0
```
[the remark this answers is not in the document]
Comment (bednar) [reply Dockerfile_build]: `Rust` has been updated to the latest version. I will prepare a `builder/README.md` file with documentation on how Kapacitor is built, including how to update builder components - Go, Rust, ...

[the remark this answers is not in the document]
Comment (bednar) [reply Dockerfile_build]: The `builder/README.md` is ready to review.

File: builder/Dockerfile_build_push.sh [hunk 2]
```diff
@@ -11,4 +11,4 @@ DOCKER_TAG="kapacitor-$(date +%Y%m%d)"
 docker build --rm=false --platform linux/amd64 -f ./Dockerfile_build -t builder:"$DOCKER_TAG" .
 docker tag builder:"$DOCKER_TAG" quay.io/influxdb/builder:"$DOCKER_TAG"
 
-docker push quay.io/influxdb/builder:"$DOCKER_TAG"
+#docker push quay.io/influxdb/builder:"$DOCKER_TAG"
```
Comment (jdstrand) [inline builder/Dockerfile_build_push.sh:14]: This should be uncommented, no?

**Review by jdstrand — CHANGES_REQUESTED** [review]

**Review by jdstrand — APPROVED** [review]
LGTM, thanks!

**Pull request merged**